common/Resources/ti.internal/extensions/process.js

```diff
@@ -44,67 +49,116 @@ function standardizeArch(original) {
+	let type;
+	let detail;
+	if (typeof options === 'string') {
+		type = options;
+	} else if (typeof options === 'object') {
+		type = options.type;
+		code = options.code;
+		detail = options.detail;
+	}
+	if (typeof warning === 'string') {
+		// TODO: make use of `ctor` arg for limiting stack traces? Can only really be used on V8
+		// set stack trace limit to 0, then call Error.captureStackTrace(warning, ctor);
+		warning = new Error(warning);
+		warning.name = type || 'Warning';
+		if (code !== undefined) {
+			warning.code = code;
 		}
-		if (typeof warning === 'string') {
-			// TODO: make use of `ctor` arg for limiting stack traces? Can only really be used on V8
-			// set stack trace limit to 0, then call Error.captureStackTrace(warning, ctor);
-			warning = new Error(warning);
-			warning.name = type || 'Warning';
-			if (code !== undefined) {
-				warning.code = code;
-			}
-			if (detail !== undefined) {
-				warning.detail = detail;
-			}
+		if (detail !== undefined) {
+			warning.detail = detail;
 		}
-		// TODO: Throw TypeError if not an instanceof Error at this point!
-		const isDeprecation = (warning.name === 'DeprecationWarning');
-		if (isDeprecation && process.noDeprecation) {
-			return; // ignore
+	}
+	// TODO: Throw TypeError if not an instanceof Error at this point!
+	const isDeprecation = (warning.name === 'DeprecationWarning');
+	if (isDeprecation && process.noDeprecation) {
+		return; // ignore
+	}
+	if (isDeprecation && process.throwDeprecation) {
+		throw warning;
+	}
+	this.emit('warning', warning);
+};
+process.env = {};
+process.execArgv = [];
+process.execPath = ''; // FIXME: What makes sense here? Path to titanium CLI here?
+process.exit = () => {
+	throw new Error('process.exit is not supported');
+};
+process.exitCode = undefined;
+process.noDeprecation = false;
+process.pid = 0;
+// FIXME: Should we try and adopt 'windowsphone'/'windowsstore' to 'win32'?
+// FIXME: Should we try and adopt 'ipad'/'iphone' to 'darwin'? or 'ios'?
+process.platform = Ti.Platform.osname;
+process.ppid = 0;
+// TODO: Add release property (Object)
+// TODO: Can we expose stdout/stderr/stdin natively?
+process.stderr = {
+	isTTY: false,
+	writable: true,
+	write: (chunk, encoding, callback) => {
+		console.error(chunk);
+		if (callback) {
+			callback();
 		}
-		if (isDeprecation && process.throwDeprecation) {
-			throw warning;
+		return true;
+	}
+};
+process.stdout = {
+	isTTY: false,
+	writable: true,
+	write: (chunk, encoding, callback) => {
+		console.log(chunk);
+		if (callback) {
+			callback();
 		}
-		this.emit('warning', warning);
+		return true;
 	}
 };
+process.title = Ti.App.name;
+process.throwDeprecation = false;
+process.traceDeprecation = false;
+process.umask = () => 0, // just always return 0
+process.uptime = () => {
+	const diffMs = Date.now() - startTime;
+	return diffMs / 1000.0; // convert to "seconds" with fractions
+};
+process.version = Ti.version,
```

Comment:
<!--
  1 failure:  common/Resources/...
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-default;
  File: common/Resources/ti.internal/extensions/process.js;
  Line: 154;
-->
- :no_entry_sign: common/Resources/ti.internal/extensions/process.js line 154 – Unexpected use of comma operator. (no-sequences)
- :warning: common/Resources/ti.internal/extensions/process.js line 154 – Expected an assignment or function call and instead saw an expression. (no-unused-expressions)